### ISSUE TITLE
Opportunity to add a seed or a random_state parameter to scipy.sparse.linalg.eigs eigsh and svds ?

### ISSUE THREAD

**Issue by FollowKenny** [scipy/scipy#9023]
scipy.sparse.linalg.eigs eigsh and svds cannot be seeded if the v0 parameter is not passed which leads to reproducibility issue in those cases. It might be useful to add a random_state or seed parameter to those functions to allow users to still set a seed without passing v0. This seed would then be used to initialize v0.

This request is related to a scikit-learn issue (https://github.com/scikit-learn/scikit-learn/issues/5545) and you can find more details there.

**Comment by rgommers** [comment]
This is the most relevant discussion I think: https://github.com/scikit-learn/scikit-learn/pull/5012#issuecomment-123601615

My first impression: 
- having any `sparse.linalg` function use random number generation without the possibility to use a seed or RandomState instance would be a bug.
- this is not a bug, because the `v0` parameter allows passing in a fixed vector rather than having ARPACK generate a random one.
- it is a usability issue however, because correctly choosing fixed `v0` without introducing any bias seems to be nontrivial.

There I would suggest adding a standard `random_state=None` keyword would be reasonable.

**Comment by FollowKenny** [comment]
All right I’ll take care of this asap after I finish the PR for Sklearn if you think it’s ok for a first timer on scipy contrib.

**Comment by pv** [comment]
It's probably ok. The only design question is that the "default" behavior is LAPACK internal RNG, not numpy's default RNG, so I guess the default value `random_state=None` should mean using the ARPACK default behavior to retain full backward compat. You may want to check the fortran code to see from what distribution the vectors are pulled from (`dgetv0/zgetv0` with `initv == .false.`, which calls [dlarnv](http://www.netlib.org/lapack/explore-3.1.1-html/dlarnv.f.html) or [zlarnv](http://www.netlib.org/lapack/explore-3.1.1-html/zlarnv.f.html) ).

**Comment by pv** [comment]
Just an addition to this discussion I remember now: IIRC, ARPACK may use its internal RNG in some cases even when v0 is specified. Whether this is really so should be checked also here first.
